src/sync/sync_manager.hpp

```diff
@@ -141,6 +147,7 @@ friend class SyncSession;
     // FIXME: Should probably be util::Logger::Level::error
     util::Logger::Level m_log_level = util::Logger::Level::info;
     SyncLoggerFactory* m_logger_factory = nullptr;
+    realm::ClientThreadListener* m_client_thread_listener;
```

Comment:
This needs to be initialized to `nullptr`. Otherwise it'll start off pointing to garbage and we'll crash when `SyncClient` finds it non-null and tries to dereference it.